lib/install/management/delete.go

```diff
@@ -238,11 +238,34 @@ func (d *Dispatcher) DeleteVCHInstances(vmm *vm.VirtualMachine, conf *config.Vir
 	var err error
 	var children []*vm.VirtualMachine
 
-	// TODO: GET CHILDREN EITHER BY FOLDER OR RP BASED ON DEPLOYMENT
-	if children, err = d.parentResourcepool.GetChildrenVMs(d.op, d.session); err != nil {
+	parentFolder, err := vmm.ParentInventoryFolder(d.op, d.session)
+	if err != nil {
 		return err
 	}
 
+	if parentFolder.Reference() == d.session.VMFolder.Reference() {
+		// get the children from the resource pool since it is an old vch
```

Comment:
It is an _old_ VCH, but I'd update the comment to state that we are using the resource pool vs. the vm-folder to minimize the number of potential VMs we are interrogating.

Reply:
+1